### ISSUE TITLE
Eager loading asset transforms doesn't work (with potential fix)

### ISSUE THREAD

**Issue by smcyr** [craftcms/cms#4931]
### Description
I try to eager load asset transforms in a matrix and with or without the eager loading, the number of queries stay the same.

I checked in the code and in the method `eagerLoadElements` of `Elements.php` line `1694`, the asset query is executed with `asArray`. In the method `populate` of `AssetQuery.php`, the `withTransforms` is skipped if `asArray` is `true`. So, if I comment `asArray()` in `Elements.php`, it's now working as intended.

Update: It seems the the culprit is this commit from 14 days ago:
https://github.com/craftcms/cms/commit/ec292df0d8b1b358d4e073ab4e32e5dead5e7b10#diff-a23f034d2592a363d6ae402cff459928

### Steps to reproduce

1. Create a matrix field with an asset field inside
2. Loop in the matrix while eager loading the asset with `withTransforms`

### Additional info

- Craft version: 3.3.2
- PHP version: 7.3.5
- Database driver & version: MySQL 5.5.5
- Plugins & versions:

**Comment by rmales** [comment]
I'm trying to debug some problems we're having with eager loading transforms and came across this issue. It seems like the fix here was reverted at some point. Now, it looks to me like because of this `withTransforms` doesn't work on nested fields, e.g. with code like:
````
{% set authors = entry.authors.with([
    ['profileImage', { withTransforms: ['avatarX1', 'avatarX2'] }],
]).all() %}
````

**Comment by gromgull** [comment]
Can confirm - it no longer works in  3.5.16 :(

**Comment by brandonkelly** [comment]
@gromgull This particular bug was fixed in September 2019. Please post a new issue with complete steps to reproduce if you’re experiencing a new issue.

**Comment by rmales** [comment]
I believe this may be fixed in 3.6 as per https://github.com/craftcms/cms/issues/6140
